File: gcj2013R1BA.hs
#!/usr/bin/env runghc
import Data.List
import Data.List.Split
main = getContents >>= mapM_ putStrLn . zipWith caseno [1..] . solve . lines
caseno i r = "Case #"++show i++": "++r
solve (c:cs) = map (toString . answer) . take (read c) . analyze $ cs ::[String]
analyze = chunksOf 2 . map (map read . words)
toString = show
answer [[a,n],m] = f a n . sort $ m
f _ n [] = n
f a n (m:ms)
  | a>m = f (a+m) (n-1) ms
  | a<=1 = n
  | otherwise = min n (f (a+a-1) (n+1) (m:ms))

File: gcj2013R1BA.py
#!/usr/bin/env python
import sys

def solve():
    A,N = map(int,sys.stdin.readline().split())
    M = map(int,sys.stdin.readline().split())
    def f(a,n,m):
        if not m or a<=1: return n
        if a>m[0]: return f(a+m[0],n-1,m[1:])
        return min(n,f(a+a-1,n+1,m))
    return f(A,N,sorted(M))

for x in range(int(sys.stdin.readline())):
    print("Case #"+str(x+1)+": "+str(solve()))
